test/ParserTests.hs

```diff
@@ -328,26 +354,26 @@ testUserDefAdHocTypes input = do
   runMain input
 
 userDefAdHocTypesSuccess = unlines $
-  [ "MyInt = let intTag = unique"
-  , "        in ( \\i -> if not i"
-  , "                   then \"MyInt must not be 0\""
-  , "                   else (intTag, i)"
-  , "           , \\i -> if dEqual (left i) intTag"
+  [ "MyInt = let wrapper = \\h -> ( \\i -> if not i"
+  , "                        then \"MyInt must not be 0\""
+  , "                   else  i"
+  , "           , \\i -> if dEqual (left i)"
   , "                   then 0"
   , "                   else \"expecting MyInt\""
   , "           )"
+  , "in wrapper (# wrapper)"
   , "main = \\i -> ((left MyInt) 8, 0)"
   ]
 
 userDefAdHocTypesFailure = unlines $
-  [ "MyInt = let intTag = unique"
-  , "        in ( \\i -> if not i"
-  , "                   then \"MyInt must not be 0\""
-  , "                   else (intTag, i)"
-  , "           , \\i -> if dEqual (left i) intTag"
+  [ "MyInt = let wrapper = \\h -> ( \\i -> if not i"
+  , "                        then \"MyInt must not be 0\""
+  , "                   else  i"
+  , "           , \\i -> if dEqual (left i)"
   , "                   then 0"
   , "                   else \"expecting MyInt\""
   , "           )"
+  , "in wrapper (# wrapper)"
   , "main = \\i -> ((left MyInt) 0, 0)"
```

Comment:
This doesn't look like it will actually fail as expected, because your code doesn't contain `abort`. Please look at @hhefesto 's latest comment on #47 for reference.